OpenRA.Mods.Common/Traits/Buildings/RallyPoint.cs

```diff
@@ -73,22 +75,34 @@ public IEnumerable<IOrderTargeter> Orders
 		public Order IssueOrder(Actor self, IOrderTargeter order, Target target, bool queued)
 		{
 			if (order.OrderID == "SetRallyPoint")
-				return new Order(order.OrderID, self, false) { TargetLocation = self.World.Map.CellContaining(target.CenterPosition), SuppressVisualFeedback = true };
+				return new Order(order.OrderID, self, false) { TargetLocation = self.World.Map.CellContaining(target.CenterPosition), SuppressVisualFeedback = true,
+					ExtraData = ((RallyPointOrderTargeter)order).SetPrimaryBuilding ? ExtraDataSetPrimaryBuilding : 0 };
 
 			return null;
 		}
 
 		public void ResolveOrder(Actor self, Order order)
 		{
 			if (order.OrderString == "SetRallyPoint")
+			{
 				Location = order.TargetLocation;
+				if (order.ExtraData == ExtraDataSetPrimaryBuilding)
+				{
+					var primaryBuildingTrait = self.Trait<PrimaryBuilding>();
+					if (primaryBuildingTrait != null)
+					{
+						primaryBuildingTrait.SetPrimaryProducer(self, true);
+					}
+				}
+			}
 		}
 
 		class RallyPointOrderTargeter : IOrderTargeter
 		{
 			public string OrderID { get { return "SetRallyPoint"; } }
 			public int OrderPriority { get { return 0; } }
 			public bool TargetOverridesSelection(TargetModifiers modifiers) { return true; }
+			public bool SetPrimaryBuilding;
```

Comment:
Property as in "public bool SetPrimaryBuilding { get; private set; }" this is to prevent someone from outside the Targeter to set the boolean while still allowing them to retrieve it.